pyfar/dsp/dsp.py

```diff
@@ -187,3 +187,112 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+        'max' - Normalize to the max or operation across all channels
+        'min' - Normalize to min or operation across all channels
+        'mean' - Normalize to mean of operation across all channels
+       The default is 'max'
+    value:
+        Normalizes to 'value' which can be a scalar or a vector with
+        a number of elements equal to channels
+        The default is 0 for norm_type='db' and 1 otherwise
+    freq_range: tuple
+        two element vector specifying upper and lower frequency bounds
+        for normalization or scalar specifying the centre frequency for
+        normalization
+        The default is (20,22000)
+    Returns
+    --------
+    normalized_signal: Signal
+        The normalized signal
+    """
+
+    # check input
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+
+    # set default values
+    if value is None:
+        value = 0 if norm_type == 'dB' else 1
+
+    # copy and transform data to the desired domain
+    if norm_type == 'time':
+        normalized_input = np.abs(signal.time.copy())
+    elif norm_type == 'abs':
+        normalized_input = np.abs(signal.freq.copy())
+    elif norm_type == 'dB':
+        normalized_input = 20 * np.log10(signal.freq.copy())
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'."))
+
+    # take logarithm of the data here instead of later?
+
+    # get bounds for normalization
+    if norm_type == 'time':
+        lim = (0, signal.n_samples)
+
+    else:
+        lim = signal.find_nearest_frequency(freq_range)
+        print(lim)
+
+        # remove 0 hz and nyquist due to normalization dependency
+        if signal.n_samples % 2:
+            lim[0] = np.max([lim[0, 1]])
+        else:
+            lim = np.clip(lim, 0)
+
+    # get values for normalization
+    if operation == 'max':
+        values = np.max(normalized_input[..., lim[0]:lim[1]], axis=-1,
+                        keepdims=True)
+    elif operation == 'mean':
+        values = np.mean(normalized_input[..., lim[0]:lim[1]], axis=-1,
+                         keepdims=True)
+    elif operation == 'rms':
+        values = np.sqrt(np.mean((normalized_input[..., lim[0]:lim[1]])**2,
+                         axis=-1, keepdims=True))
+    else:
+        raise ValueError(("operation must be 'abs', 'mean' or 'rms'."))
+
+    # manipulate values
+    if channelwise == 'each':
+        pass
+    elif channelwise == 'max':
+        values = np.max(values)
+    elif channelwise == 'min':
+        values = np.min(values)
+    elif channelwise == 'mean':
+        values = np.mean(values)
+    else:
+        raise ValueError(("channelwise must be 'each', 'max', 'min' or 'mean'."
+                          ))
+    # apply normalization
+    normalized_input = (normalized_input / values) * value
+
+    # replace input with normalized_input
+    normalized_signal = signal.copy()
+    normalized_signal.time = normalized_input
```

Comment:
This might assign a log. magnitude spectrum to `.time`. I would suggest to delete 292 and use this for 296:
`normalized_signal.time = signal.time.copy() / values * value`

If you do this you have to de-logarithmize value if it is in dB, i.e., if `norm_type='dB'`